java/memory/src/main/java/org/apache/arrow/memory/BoundsChecking.java

```diff
@@ -17,6 +17,12 @@
 
 package org.apache.arrow.memory;
 
+/**
+ * Configuration class to determine if bounds checking should be turned on or off.
+ *
+ * <p>Bounds checking is turned on by setting the system property
```

Comment:
this looks in conflict with the code.

Setting the property seems to turn off bounds checking, while setting to false(which is the default) turns it on.

Reply:
Good catch, don't know how I missed that.  reworded.  If I read it correctly now it seems like both system properties need to be set to true to turn off bounds checking, this seems like odd behavior (requiring a deprecated parameter)?

Reply:
definitely odd :) the drill parameter should be marked deprecated (and have a default?)

Reply:
ARROW-4851